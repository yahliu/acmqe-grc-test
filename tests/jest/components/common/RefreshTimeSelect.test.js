/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/* Copyright (c) 2020 Red Hat, Inc. */
'use strict'

import React from 'react'
import RefreshTimeSelect from '../../../../src-web/components/common/RefreshTimeSelect'
import renderer from 'react-test-renderer'

const refreshControl = {
  reloading: false,
  refreshCookie: 'grc-refresh-interval-cookie',
  timestamp: 'Tue Sep 24 2019 16:07:17 GMT-0400 (Eastern Daylight Time)',
  startPolling: jest.fn(),
  stopPolling: jest.fn()
}
const refreshValues = [5,10,30,60,300,1800,0]

describe('RefreshTimeSelect component', () => {
  it('renders as expected without data', () => {
    const component = renderer.create(
      <RefreshTimeSelect
        locale = {'en-US'}
        refreshValues = {[]}
        refreshControl = {{}}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('renders as expected with data', () => {
    const component = renderer.create(
      <RefreshTimeSelect
        locale = {''}
        refreshValues = {refreshValues}
        refreshControl = {refreshControl}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('renders as expected on change', () => {
    const component = renderer.create(
      <RefreshTimeSelect
        locale = {''}
        refreshValues = {refreshValues}
        refreshControl = {refreshControl}
      />
    )
    component.root.findByProps({className: 'selection'}).props.onChange({selectedItem: {pollInterval: 0}})
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('renders loading when reloading=true', () => {
    refreshControl.reloading = true
    const component = renderer.create(
      <RefreshTimeSelect
        locale = {''}
        refreshValues = {refreshValues}
        refreshControl = {refreshControl}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
