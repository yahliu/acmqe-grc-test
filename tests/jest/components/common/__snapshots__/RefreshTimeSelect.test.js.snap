// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RefreshTimeSelect component renders as expected on change 1`] = `
<div
  className="refresh-time-selection"
>
  <div
    aria-label="Refresh"
    className="button"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex={0}
    title="Refresh"
  >
    <svg
      className="button-icon"
    >
      <use
        href="#diagramIcons_autoRefresh"
      />
    </svg>
  </div>
  <div
    aria-label="Choose refresh"
    className="bx--dropdown selection bx--list-box"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="listbox"
    tabIndex={null}
  >
    <div
      aria-controls={null}
      aria-expanded={false}
      aria-haspopup={true}
      aria-label="Choose refresh"
      aria-owns={null}
      className="bx--list-box__field"
      data-toggle={true}
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
      type="button"
    >
      <span
        className="bx--list-box__label"
        htmlFor="downshift-2-input"
      >
        Refresh every {0}s
      </span>
      <div
        className="bx--list-box__menu-icon"
      >
        <svg
          alt="Open menu"
          aria-label="Open menu"
          fillRule="evenodd"
          focusable="false"
          height="5"
          role="img"
          viewBox="0 0 10 5"
          width="10"
        >
          <title>
            Open menu
          </title>
          <path
            d="M0 0l5 4.998L10 0z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`RefreshTimeSelect component renders as expected with data 1`] = `
<div
  className="refresh-time-selection"
>
  <div
    aria-label="Refresh"
    className="button"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex={0}
    title="Refresh"
  >
    <svg
      className="button-icon"
    >
      <use
        href="#diagramIcons_autoRefresh"
      />
    </svg>
  </div>
  <div
    aria-label="Choose refresh"
    className="bx--dropdown selection bx--list-box"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="listbox"
    tabIndex={null}
  >
    <div
      aria-controls={null}
      aria-expanded={false}
      aria-haspopup={true}
      aria-label="Choose refresh"
      aria-owns={null}
      className="bx--list-box__field"
      data-toggle={true}
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
      type="button"
    >
      <span
        className="bx--list-box__label"
        htmlFor="downshift-1-input"
      >
        Refresh every {0}s
      </span>
      <div
        className="bx--list-box__menu-icon"
      >
        <svg
          alt="Open menu"
          aria-label="Open menu"
          fillRule="evenodd"
          focusable="false"
          height="5"
          role="img"
          viewBox="0 0 10 5"
          width="10"
        >
          <title>
            Open menu
          </title>
          <path
            d="M0 0l5 4.998L10 0z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`RefreshTimeSelect component renders as expected without data 1`] = `
<div
  className="refresh-time-selection"
>
  <div
    aria-label="Refresh"
    className="button"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex={0}
    title="Refresh"
  >
    <svg
      className="button-icon"
    >
      <use
        href="#diagramIcons_autoRefresh"
      />
    </svg>
  </div>
  <div
    aria-label="Choose refresh"
    className="bx--dropdown selection bx--list-box"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="listbox"
    tabIndex={null}
  >
    <div
      aria-controls={null}
      aria-expanded={false}
      aria-haspopup={true}
      aria-label="Choose refresh"
      aria-owns={null}
      className="bx--list-box__field"
      data-toggle={true}
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
      type="button"
    >
      <span
        className="bx--list-box__label"
        htmlFor="downshift-0-input"
      >
        Choose refresh
      </span>
      <div
        className="bx--list-box__menu-icon"
      >
        <svg
          alt="Open menu"
          aria-label="Open menu"
          fillRule="evenodd"
          focusable="false"
          height="5"
          role="img"
          viewBox="0 0 10 5"
          width="10"
        >
          <title>
            Open menu
          </title>
          <path
            d="M0 0l5 4.998L10 0z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`RefreshTimeSelect component renders loading when reloading=true 1`] = `
<div
  className="refresh-time-selection"
>
  <div
    aria-live="assertive"
    className="bx--loading bx--loading--small"
  >
    <svg
      className="bx--loading__svg"
      viewBox="-75 -75 150 150"
    >
      <title>
        Active loading indicator
      </title>
      <circle
        className={null}
        cx="0"
        cy="0"
        r="37.5"
      />
    </svg>
  </div>
  <div
    aria-label="Choose refresh"
    className="bx--dropdown selection bx--list-box"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="listbox"
    tabIndex={null}
  >
    <div
      aria-controls={null}
      aria-expanded={false}
      aria-haspopup={true}
      aria-label="Choose refresh"
      aria-owns={null}
      className="bx--list-box__field"
      data-toggle={true}
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
      type="button"
    >
      <span
        className="bx--list-box__label"
        htmlFor="downshift-3-input"
      >
        Disable refresh
      </span>
      <div
        className="bx--list-box__menu-icon"
      >
        <svg
          alt="Open menu"
          aria-label="Open menu"
          fillRule="evenodd"
          focusable="false"
          height="5"
          role="img"
          viewBox="0 0 10 5"
          width="10"
        >
          <title>
            Open menu
          </title>
          <path
            d="M0 0l5 4.998L10 0z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
